### PR TITLE
adding-new-qa&testing-ques

### PR DESCRIPTION
**Pull request by geekymehta**
## Summary of changes
Added 8 new testing ques in https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/src/data/quality-assurance-quiz.ts

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/quality-assurance-quiz.ts [hunk 1]
```diff
@@ -1,4 +1,92 @@
...
+  },
+  {
+    Question: "Which software development approach often emphasizes the use of unit testing?",
+    Answer: 
+      "Test-Driven Development (TDD).",
+    Distractor1: "Waterfall model.",
+    Distractor2: "Agile development.",
+    Distractor3: "Scrum methodology.",
+    Explanation: 
+      "TDD involves writing unit tests before implementing the code, making unit testing a fundamental practice.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:32]:
```suggestion
      "TDD (Test-Driven Development) involves writing unit tests before implementing the code, making unit testing a fundamental practice.",
```

File: src/data/quality-assurance-quiz.ts [hunk 2]
```diff
@@ -1,4 +1,92 @@
...
+    Link: "https://www.freecodecamp.org/news/unit-tests-explained/"
+  },
+  {
+    Question: "In JavaScript, which assertion library is commonly used for writing unit tests?",
+    Answer: 
+      "Chai.",
+    Distractor1: "Mocha.",
+    Distractor2: "Jasmine.",
+    Distractor3: "Jest.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:41]: the periods aren't necessary here.

Also, a lot of these options like Jest are commonly used for unit tests to.

```suggestion
      "Chai",
    Distractor1: "Yii",
    Distractor2: "Spock",
    Distractor3: "NUnit",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by geekymehta** [comment]
@jdwilkin4, i've made those changes.

**Review by jdwilkin4 — APPROVED** [review]
Congrats on another contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**